platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1888,43 +1946,22 @@ private enum ProperMethodUsage {
             boolean isAllowed(ConfigurationInternal configuration) {
                 return configuration.isCanBeConsumed();
             }
-
-            @Override
-            boolean isDeprecated(ConfigurationInternal configuration) {
```

Comment:
I think we still need to check for depreactions. 

Our method guards (`failOnInvalidUsage`) should have the behavior:
1. if conf is disabled for role fail
2. if conf is deprecated for rile, emit deprecation

Since this method can emit both a failure and deprecation, I think we should choose a different name than `failOnInvalidUsage` that doesn't include `fail`